python/tester/utils.py

```diff
@@ -121,18 +130,11 @@ def make_hsc_id(group_num, snap):
         ``group`` and ``snap``, in the form it appears in HSC headers.
     exposure_num : `int`
         The exposure ID genereated by Middleware from ``exposure_header``.
-
-    Notes
-    -----
-    The current implementation allows up to 1000 group numbers per day.
-    It can overflow with ~60 calls to upload.py on the same day or
-    upload_hsc_rc2.py with a large N_GROUPS.
     """
-    # This is a bit too dependent on how group_num is generated, but I want the
-    # group number to be discernible even after compressing to 8 digits.
-    night_id = (group_num // 100_000) % 2020_00_00     # Always 5 digits
-    run_id = group_num % 100_000                       # Up to 5 digits, but usually 2-3
-    exposure_id = (night_id * 1000) + (run_id % 1000)  # Always 8 digits
+    # Use hash to distribute evenly over space of allowed IDs -- raw
+    # group ID has many digits that are almost always 0.
+    # hash returns an int64 on most systems, but HSC max is much smaller.
+    exposure_id = hash(group_num) % max_exposure["HSC"]
```

Comment:
The three problems with this are that it loses the monotonicity (which is not required but nice to have), it loses the correlation with the time of the test (also not required but nice to have), and it has potential collisions which cannot be predicted.

How about `((YYYY-2020)*1200+MMDD)*1000 + group_id % 1000`, which is good until the end of Oct 2037, by which time we had better be using real LSSTCam for testing?

Reply:
Where does the factor of 1200 come from? I guess the earlier algorithm was essentially `(YYYY-2020)*10_000+MMDD`...

Though I do like it -- the date term is limited to 4 digits through 2027, and if we reserve 4 digits for the group ID then we're much less likely to overflow.

Reply:
1200 is from 12 months per year and a two digit shift to clear the days.  If you want to have 9999 groups (4 digits), though, then you'd have to change the epoch year because you're limited to 1.75 years before you hit 21474800.